Tests/WeaverTests.cs

```diff
@@ -156,7 +170,18 @@ public void UniquePropertyTypeButExplicitMethodName_WithIsEmittedWithLongName()
     public void OriginalWithMethodIsRemoved()
     {
         var type1 = assembly.GetType("AssemblyToProcess.PrimitiveValues");
-        Assert.False(type1.GetMethods().Any(m => m.Name == "With" && m.IsGenericMethod));
+        // check for the generic With method
+        Assert.False(type1.GetMethods().Any(m =>
+            m.IsPublic &&
+            m.Name == "With" &&
+            m.IsGenericMethod &&
+            m.GetParameters().Length == 1));
+        // check for a With method with Object parameter
+        Assert.False(type1.GetMethods().Any(m => 
+            m.IsPublic &&
+            m.Name == "With" && 
+            m.GetParameters().Length == 1 &&
+            m.GetParameters()[0].ParameterType == typeof(object)));
```

Comment:
Ah, that's my mistake then. The first version was using `object`, and then I switched to generic methods. I'll fix the tests, or you may submit a PR ;)